### ISSUE TITLE
o-header: make esc key close menus which were opened by hover or focus - ID: DAC_Content_On_Hover_01

### ISSUE THREAD

**Issue by JakeChampion** [Financial-Times/origami#549]
# Issue
This is from page 167 of the [audit report](https://drive.google.com/file/d/17wq6b0CIX41RA4muBFDzUDxE0-qzHBIH/view)

Issue can be found on: <https://www.ft.com/__origami/service/build/v3/demo?component=o-header@9.0.1&demo=mega-menu&system_code=origami-repo-data&brand=master>

When users hover or focus the links which expand to reveal content below, the menu cannot be closed without moving the mouse (using the Esc key on the keyboard). This issue may affect low vision users who use the mouse to track focus on the page.

# Proposed Solution:
Ensure that users are able to close the expanding menu without having to use the mouse.
